etc/helm/pachyderm/templates/pachd/deployment.yaml

```diff
@@ -224,12 +255,27 @@ spec:
           value: {{ .Values.pachd.securityContext.enabled | quote }}
         - name: UNPAUSED_MODE
           value: "full"
+        {{- if .Values.pachd.activateEnterpriseMember }}
         - name: ENTERPRISE_MEMBER
           value: {{ .Values.pachd.activateEnterpriseMember | quote }}
+        - name: ENTERPRISE_SERVER_ADDRESS
+          value: {{ required "To register a pachd instance with an enterprise server, the enterprise server's address must be provided." .Values.pachd.enterpriseServerAddress | quote }}
+        - name: ENTERPRISE_SERVER_TOKEN
+          {{- if .Values.pachd.enterpriseRootTokenSecretName }}
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.pachd.enterpriseRootTokenSecretName | quote}}
+              value: enterprise-root-token
+          {{- else }}
+          value: {{ required "To register this pachd as an enterprise member, the enterprise root token must be provided." .Values.pachd.enterpriseRootToken | quote }}
+          {{- end }}
+        {{- end }}
         {{ if .Values.global.postgresql.identityDatabaseFullNameOverride }}
         - name: IDENTITY_SERVER_DATABASE
           value: {{ .Values.global.postgresql.identityDatabaseFullNameOverride }}
         {{ end }}
+        - name: UPGRADE_NO_OP
+          value: {{ randAlphaNum 32 }}
```

Comment:
Customers definitely want this to happen (automatically restart pachd when the secrets change), but this will break argocd, because it just does "helm template" in a loop, and this will make every iteration look to argo like a new change.

I think we'll have to hash each of the secrets we generate and put them here instead.  That way only a legitimate edit of a relevant field in values.yaml triggers this automatic restart.  (Another example; if we set the resources on console to a new value, pachd shouldn't need to restart.)  

Great idea for a feature though.  Let's do it in another PR soon?

Reply:
What if we wrap this env-var in an `isUpgrade` conditional? I think we should chat about the ArgoCD stuff more - as a part of the current approach helm will spit out new random values in the secrets each run if left unspecified- will that break an ArgoCD workflow?